integrations/spring-boot/src/main/java/one/microstream/integrations/spring/boot/types/storage/StorageBeanFactory.java

```diff
@@ -0,0 +1,158 @@
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+@Component
+public class StorageBeanFactory implements ApplicationContextAware, BeanDefinitionRegistryPostProcessor
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StorageBeanFactory.class);
+
+    private ApplicationContext applicationContext;
+
+    private final Set<Class<?>> storageClasses = new HashSet<>();
+
+    public Object createRootObject()
+    {
+        LOGGER.debug("Creating Spring bean for @Storage annotated class");
+        final StorageManager storageManager = this.applicationContext.getBean(StorageManager.class);
+        Object root = storageManager.root();
+
+        if (Objects.isNull(root))
+        {
+            root = XReflect.defaultInstantiate(storageClasses.iterator()
+                                                       .next());
+            LOGGER.debug(String.format("Created Root object from class %s", root.getClass().getName()));
+            storageManager.setRoot(root);
+            storageManager.storeRoot();
+        }
+
+        this.applicationContext
+                .getAutowireCapableBeanFactory()
+                .autowireBean(root);
+
+        this.processInitializers(storageManager);
+
+        return root;
+    }
+
+    private void processInitializers(final StorageManager storageManager)
+    {
+        final String[] namesForType = applicationContext.getBeanNamesForType(StorageManagerInitializer.class);
+        Arrays.stream(namesForType)
+                .map(name -> applicationContext.getBean(name))
+                .map(StorageManagerInitializer.class::cast)
+                .forEach(initializer -> initializer.initialize(storageManager));
+
+    }
+
+    @Override
+    public void setApplicationContext(final ApplicationContext applicationContext) throws BeansException
+    {
+        this.applicationContext = applicationContext;
+    }
+
+    @Override
+    public void postProcessBeanDefinitionRegistry(final BeanDefinitionRegistry beanDefinitionRegistry) throws BeansException
+    {
+        for (String definitionName : beanDefinitionRegistry.getBeanDefinitionNames())
+        {
+
+            final BeanDefinition beanDefinition = beanDefinitionRegistry.getBeanDefinition(definitionName);
+            if (beanDefinition.getBeanClassName() != null)
+            {
+                try
+                {
+                    // Don't use generics as that will not work with the Supplier that provides the Root object further in this method
+                    final Class clazz = Class.forName(beanDefinition.getBeanClassName());
```

Comment:
<img width="645" alt="image" src="https://user-images.githubusercontent.com/17253582/181708057-0b05e35d-f645-426b-b0bb-939d739635a1.png">
There are warnings here, is it ok?